src/desktop/apps/fair/fairRedirection.js

```diff
@@ -0,0 +1,56 @@
+import { DISABLE_FAIRS_UPDATE_SLUGS, ENABLE_FAIRS_UPDATE } from "../../config"
+
+export const redirectFairRequests = (req, res, next) => {
+  // TODO: some time after launch of the updated fairs experience
+  // this entire function can be reduced to the res.redirect,
+  // sans conditional logic
+  if (shouldRedirectToNewFairPage(req)) {
+    res.redirect(301, redirectPath(req, res))
+  } else {
+    next()
+  }
+}
+
+const shouldRedirectToNewFairPage = req => {
+  if (isNewFairPageDisabledForThisFair(req)) return false
+  if (shouldUserSeeNewFairPage(req)) return true
+
+  return false
+}
+
+const isNewFairPageDisabledForThisFair = req => {
+  return !!DISABLE_FAIRS_UPDATE_SLUGS?.split(/\s*,\s*/).includes(req.params.id)
+}
+
+const shouldUserSeeNewFairPage = req => {
+  const userHasLabFeature = !!req.user?.hasLabFeature("Fairs experience update")
+  const featureIsEnabledSitewide = ENABLE_FAIRS_UPDATE
+
+  return userHasLabFeature || featureIsEnabledSitewide
+}
+
+const redirectPath = (req, res) => {
+  const defaultPath = `/fair/${req.params.id}`
```

Comment:
Wondering about this! Is this accurate? I thought that going to `/:id` is essentially, loading a profile whose owner is a fair. In which case, is it guaranteed that the handle of the profile will be equivalent to the fair slug in all cases?

Reply:
Good q! Lemme try to confirm.

Reply:
They are the same.... 88% of the time 😓 

I'll add the logic to pull the profile's owner's slug instead.